### ISSUE TITLE
Typespecs aren't quite right

### ISSUE THREAD

**Issue by rzane** [infinitered/authority_ecto#20]
So, the first problem is that this package is expecting a type `t` to be defined on both the token schema and user schema. Since Ecto.Schema doesn't automatically generate a type, this assumption isn't a very safe one.

So, I defined a type on each schema:

```elixir
@type t :: %__MODULE__{}
```

But then, I get some violations:

```
lib/authority_ecto/templates/registration.ex:1: Function change_user/0 has no local return
lib/authority_ecto/templates/registration.ex:210: The call 'Elixir.MyApp.Accounts':change_user('nil',#{}) breaks the contract ('Elixir.MyApp.Accounts.User':t() | 'Elixir.MyApp.Accounts.Token':t(),map()) -> {'ok','Elixir.Ecto.Changeset':t()} | auth_failure()
```

**Comment by danielberkompas** [comment]
We could update the typespecs to use `%schema{}` instead of `schema.t()`. Might help.

**Comment by danielberkompas** [comment]
Fixed by 8a06269

**Comment by danielberkompas** [comment]
I released a new version with this fix: `0.1.2`
